### PR TITLE
 docs: javadocs config for api-client

### PR DESCRIPTION
**Pull request by colinhicks**
This is a stacked PR. Please review [latest commits](https://github.com/confluentinc/ksql/pull/5581/files/eca3fd8c53750296f99aed0cd80d614361eeab8b..3e91103f9b04fadd8d389c3f676a63ed538f2936) only.

Here is a staged preview:
https://docs.ksqldb.io/en/java-client-docs_javadocs_built/developer-guide/ksqldb-clients/java-client/api/io/confluent/ksql/api/client/package-summary.html

- [x] javadocs configuration
- [x] analytics integration
- [x] link to API docs from Java Client docs page
- [x] attempt to integrate with docs site chrome
- [ ] document workflow

### CODE REVIEW

**Comment by colinhicks** [comment]
One challenge is that our docs site CI requires that all docs sources are committed. In other words, we can't run `mvn -pl ksqldb-api-client javadocs:javadocs` on that build server. We also do not actually want to commit the api docs artifacts where possible. So I've added their output path to the root gitignore.

My thinking so far is:
- ksql-db-release builds the docs only in release jobs.
- generated api docs are added to the index (just before a given release commit/tag) via `git add -f docs/developer-guide/ksqldb-clients/java-client/api`.
- built javadocs are tracked in GH for release branches, so that the docs build can consume them.
- fixes to the api docs can be committed manually to the release branches. They will need to be added with the force flag, as above.

**Comment by vcrfxia** [comment]
Looks great @colinhicks -- thanks a bunch! 

> We also do not actually want to commit the api docs artifacts where possible. So I've added their output path to the root gitignore.

How come we don't want to commit the API docs? Is it just bad practice to commit generated artifacts, or something else?

> ksql-db-release builds the docs only in release jobs.

Does this mean local builds will not build docs (by default) either? I was going to ask how much this additional step slows down the build, but it sounds like that might not be a concern.

**Comment by colinhicks** [comment]
Thanks for taking a look, @vcrfxia. Yeah, I was suggesting we avoid checking in the generated files in most cases so that the javadoc comments in class files are the single source of truth. In practice, generated files can add noise to PR reviews.

> Does this mean local builds will not build docs (by default) either?

Yeah they wouldn't build by default unless you invoke the `site` Maven lifecycle or the `javadocs` goal directly.

**Comment by colinhicks** [comment]
Made some configuration tweaks and integrated with the livery 
image

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 1]
```diff
@@ -0,0 +1,473 @@
+---
+layout: page
+title: Java Client for ksqlDB
+tagline: Java client for ksqlDB
+description: Send requests to ksqlDB from your Java app
+keywords: ksqlDB, java, client
+---
+
+ksqlDB ships with a lightweight Java client that allows you to easily send requests to a ksqlDB server
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-clients/java-client.md:9]:
```suggestion
ksqlDB ships with a lightweight Java client that enables send requests easily to a ksqlDB server
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+from within your Java application, as an alternative to using the [REST API](../api.md).
+The client currently supports pull and push queries as well as inserting new rows of data into existing ksqlDB streams.
+Soon the client will also support persistent queries and admin operations such as listing streams, tables, and topics.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-clients/java-client.md:12]: Generally, we don't want to make promises about future features in docs, but It's probably fine in this case.

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+from within your Java application, as an alternative to using the [REST API](../api.md).
+The client currently supports pull and push queries as well as inserting new rows of data into existing ksqlDB streams.
+Soon the client will also support persistent queries and admin operations such as listing streams, tables, and topics. 
+
+!!! tip
+    [View the Java client API documentation](api/index.html)
+
+The client sends requests to the recently added HTTP2 server endpoints: pull and push queries are served by
+the [`/query-stream` endpoint](TODO), and inserts are served by the [`/inserts-stream` endpoint](TODO).
+Consequently, the client should only be used with ksqlDB deployments on version 0.10.0 or newer.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-clients/java-client.md:19]:
```suggestion
Use the client only with ksqlDB deployments on version 0.10.0 or newer.
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+    }).exceptionally(e -> {
+      System.out.println("Request failed: " + e);
+      return null;
+    });
+```
+
+### Synchronous Usage ###
+
+To consume records one-at-a-time in a synchronous fashion, use the `poll()` method on the query result object.
+If `poll()` is called with no arguments, `poll()` blocks until a new row becomes available or the query is terminated.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-clients/java-client.md:212]:
```suggestion
If `poll()` is called with no arguments, blocks until a new row becomes available or the query is terminated.
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+   */
+  BatchedQueryResult executeQuery(String sql);
+  
+  ...
+  
+}
+```
+
+This method is suitable for both pull queries and for terminating push queries, for example, queries that have a `LIMIT` clause).
+For non-temrinating push queries, use [the `streamQuery()` method](./stream-query.md) instead.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-clients/java-client.md:260]:
```suggestion
For non-terminating push queries, use [the `streamQuery()` method](./stream-query.md) instead. 
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+
+```java
+String pushQuery = "SELECT * FROM MY_STREAM EMIT CHANGES;";
+StreamedQueryResult streamedQueryResult = client.streamQuery(pushQuery).get();
+
+String queryId = streamedQueryResult.queryID();
+client.terminatePushQuery(queryId).get();
+```
+
+And here's an analogous example for terminating a push query issued by using the `executeQuery()` method:
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-clients/java-client.md:321]:
```suggestion
Here's an analogous example for terminating a push query issued by using the `executeQuery()` method:
```

File: docs/developer-guide/ksqldb-clients/java-client.md [hunk 7]
```diff
@@ -0,0 +1,473 @@
...
+KsqlObject row = new KsqlObject()
+    .put("ROWKEY", "k1")
+    .put("ORDER_ID", 12345678L)
+    .put("PRODUCT_ID", "UAC-222-19234")
+    .put("USER_ID", "User_321");
+
+client.insertInto("ORDERS", row).get();
+```
+
+Tutorial Examples
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-clients/java-client.md:374]: These are great!

**Comment by colinhicks** [comment]
Thanks for the review, @JimGalasyn. @vcrfxia, since Jim's suggestions apply to your stacked commits under review in the other PR, where do you think we should apply them?

**Comment by vcrfxia** [comment]
> Thanks for the review, @JimGalasyn. @vcrfxia, since Jim's suggestions apply to your stacked commits under review in the other PR, where do you think we should apply them?

I'll make the changes in my PR -- works out since I haven't actually merged yet :)

**Pull request merged**